syntax = "proto3";

package authorization;

service AuthorizationService {
    rpc SignUp(SignUpRequest) returns (SignUpResponse);
    rpc SignIn(SignInRequest) returns (SignInResponse);
    rpc ValidToken(ValidTokenRequest) returns (ValidTokenResponse);
}

message SignUpRequest {
    string email = 1;
    string password = 2;
    optional string name = 3;
}

message SignUpResponse {
    bool success = 1;
    string message = 2;
    optional string token = 3;
}

message SignInRequest {
    string login = 1;
    string password = 2;
}

message SignInResponse {
    bool success = 1;
    string message = 2;
    optional string token = 3;
}

message ValidTokenRequest {
    string login = 1;
    string token = 2;
}

message ValidTokenResponse {
    bool success = 1;
}
